go/vt/sqlparser/parse_date.go

```diff
@@ -0,0 +1,47 @@
+package sqlparser
+
+import (
+	"time"
+
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+var dateFormats = []string{"2006-01-02", "06-01-02", "20060102", "060102"}
+var datetimeFormats = []string{"2006-01-02 15:04:05", "06-01-02 15:04:05", "20060102150405", "060102150405"}
```

Comment:
These formats are based on https://dev.mysql.com/doc/refman/8.0/en/date-and-time-literals.html. I haven't expanded `TIME` further because that format isn't really compatible with Go at all, especially when multiple days are involved.

The `TIME` type is more a duration in a way than a timestamp, although you can also do comparisons in MySQL with a `DATE` or `DATETIME`. But expanding that is something separate based on need.

Reply:
I've added more complete `TIME` parsing in https://github.com/vitessio/vitess/pull/10921/commits/57046c394208f67ffd1ad0b65fea4441929a90b1